src/app/service/users/users.service.ts

```diff
@@ -42,17 +43,17 @@ export class UsersService {
     ))
   }
 
-  isUserSignedIn() {
+  isUserSignedIn(): string | null | undefined {
     return this.localStorage?.getItem("isLoggedIn")
   }
 
-  signOutUser() {
+  signOutUser(): void {
     localStorage.clear()
-    this.isLoogedInSubject.next(null)
+    this.isLoggedInSubject.next(null)
     this.router.navigate(["/login"])
   }
 
-  getUser() {
+  getUser(): Observable<User> | null | undefined {
     var id = this.localStorage?.getItem("id")
     var body = {
       token: this.localStorage?.getItem("token")
```

Comment:
also here you dont actually have to subscribe to it inside the service, bad practice. Instead just use this method in app component or wherever you call getUser() method and then inside there you can subscribe and use the subject to update the user$ observable